guess-the-country/country-frontend/src/App.vue

```diff
@@ -193,10 +193,18 @@ export default {
   async created() {
     this.getMessage();
     this.getStreetview();
+    this.getSequenceMode();
```

Comment:
Das müsste zuerst stattfinden und auch abgeschlossen sein, bevor die vorherigen beiden Funktionen aufgerufen werden, da deren Rückgabewert u.U. vom seuqence_mode abhängt.

Ich würde das so machen:

```node
async created() {
let urlParams = new URLSearchParams(window.location.search.substring(1));
      if (urlParams.has('modus')) {
        this.sequence_mode = urlParams.get('modus');
      }
     if (urlParams.has('num_rounds')) {
       ...
     }
     // ... alle anderen parameter parsen ...

    this.getMessage()
    this.getStreetview()
}
```

Reply:
Mir ist beim Schreiben übrigens eingefallen, wo wir das im `visual-inspection` Use Case doch auch so machen: https://github.com/XAI-Demonstrator/xai-demonstrator/blob/master/visual-inspection/inspection-frontend/src/components/ExplainInspection.vue#L42

Reply:
Ich habe mir notiert, dass die folgenden Paramter mit der URL übergeben werden: ``Player.unique_id``, ``modus``, ``image_set``, ``num_rounds``, ``embedded``.
Habe ich da etwas übersehen?

- Player.unique_id: In Zeile 115 in App.vue wird bereits einer player_id aus der URL über "player" zugewiesen. 
- image_set: beinhaltet das die 4 bzw 8 Bilder? In getStreetview() würden dann doch das zugewiesene Bild überschrieben werden?
- embedded: das ist mir unklar
- num_rounds und modus: sind klar

Reply:
```python 
   await this.parseURlParameters();
```
Würde das so auch funktionieren. Das scheint mir nämlich etwas aufgeräumter zu sein.

Reply:
> Mir ist beim Schreiben übrigens eingefallen, wo wir das im `visual-inspection` Use Case doch auch so machen: https://github.com/XAI-Demonstrator/xai-demonstrator/blob/master/visual-inspection/inspection-frontend/src/components/ExplainInspection.vue#L42

Tatsächlich machen wir es sogar schon in gtc: https://github.com/XAI-Demonstrator/xai-demonstrator/blob/2009a97dcc41348c2f2feb222e55df27390fcb27/guess-the-country/country-frontend/src/App.vue#L110

Reply:
Ah, sieh an, sehr gut!

Ich würde glaube ich alles über `computed()`-Variablen lösen. Das ist die sicherste und aufgeräumteste Variante, weil die Variable dann nicht in `data()` und an noch irgendwelchen Stellen gesetzt wird, dazu noch in schwer vorhersehbarer Reihenfolge, sondern nur an einem Ort.

Und zwar würde ich das so machen:

```node
computed: {
   numOfRounds() {
      const uri = window.location.search.substring(1);
      let params = new URLSearchParams(uri);
      if params.has("num_of_rounds") {
          return params.get("num_of_rounds")
     } else {
          return JSON.parse(process.env.VUE_APP_NR_OF_ROUNDS) 
     }
   }
}
```

Wenn dir das zu viel Code-Duplikation ist, kannst du auch eine Hilfs-Methode machen:

```node
getURLParameter(name, defaultValue) {
      const uri = window.location.search.substring(1);
      let params = new URLSearchParams(uri);
      if params.has(name) {
          return params.get(name)
     } else {
          return defaultValue
     }
}
```

und dann

```node
numOfRounds() {
    return getURLParameter("num_of_rounds", JSON.parse(process.env.VUE_APP_NR_OF_ROUNDS))
}
```